### PR TITLE
Use FunctionKeyword for computed_value of timing function

### LINKED ISSUE #15086: Keyword values of timing function should not be computed to bezier function
Per [spec](https://drafts.csswg.org/css-transitions-1/#single-transition-timing-function), the computed value of timing function is same as specified value, and in both Gecko and Blink, the keyword values are perserved in computed value.

Test the following code:
```html
<!DOCTYPE html>
<div id="x"></div>
<script>
alert(getComputedStyle(x).animationTimingFunction);
x.style.animationTimingFunction = 'ease';
alert(getComputedStyle(x).animationTimingFunction);
</script>
```

In Gecko and Blink, both alerts show `ease`, while in Stylo, the first shows `ease` while the second shows `cubic-bezier(0.25, 0.1, 0.25, 1)`, which is incorrect.

cc @hiikezoe

### PR DESCRIPTION
**Pull request by hiikezoe**
https://bugzilla.mozilla.org/show_bug.cgi?id=1358330

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15086

- [X] These changes do not require tests because mozilla-central has test cases.  There might be some test cases in servo tree.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16640)

### CODE REVIEW

File: components/style/animation.rs [hunk 1]
```diff
@@ -367,7 +372,10 @@ impl PropertyAnimation {
                     out = 1.0;
                 }
                 out
-            }
+            },
+            TransitionTimingFunction::Keyword(_) => {
+                unreachable!("Keyword function should not appear")
```
Comment (nox) [inline components/style/animation.rs]: You probably want `panic!` here, `unreachable!` AFAIK is for more obvious code, not for invariants maintained elsewhere.

Comment (hiikezoe) [inline components/style/animation.rs]: Thank you!

**Comment by hiikezoe** [comment]
OK. All of failure tests for timing-function are wrong. 
Those are also marked as 'FAIL' on mozilla-central. Sigh.

https://hg.mozilla.org/mozilla-central/file/2cca333f546f/testing/web-platform/meta/css/css-transitions-1/transition-001.html.ini
https://hg.mozilla.org/mozilla-central/file/2cca333f546f/testing/web-platform/meta/css/css-transitions-1/transition-timing-function-001.html.ini

**Comment by hiikezoe** [comment]
I'd like to fix the test cases in mozilla-central because it will be synced with upstream.
Filed; https://bugzilla.mozilla.org/show_bug.cgi?id=1360410

**Comment by jdm** [comment]
@hiikezoe Servo's web-platform-tests also syncs with upstream.

**Comment by hiikezoe** [comment]
Thank you @jdm.  Xidorn also notice me that we should not touch tests added by someone else, so I leave the failure tests as they are here.

@bors-servo try

**Comment by jdm** [comment]
There's nothing wrong with touching tests if we're actually confident that the changes are correct.

**Comment by hiikezoe** [comment]
OK,  I will do change tests with PR next time. Thank you!

**Comment by hiikezoe** [comment]
Coping .ini files from mozilla-central were slightly different..
@bors-servo try

**Pull request merged**